lib/utils/visFileArchive.hpp

```diff
@@ -8,12 +8,27 @@
 #include <highfive/H5DataSet.hpp>  // for DataSet
 #include <highfive/H5DataType.hpp> // for DataType
 #include <highfive/H5File.hpp>     // for File
+#include <hdf5.h>                  // H5Dwrite, etc.
 #include <map>                     // for map
 #include <memory>                  // for allocator, unique_ptr
 #include <stddef.h>                // for size_t
 #include <string>                  // for string
 #include <vector>                  // for vector
 
+template<typename T>
+static inline void write_hyperslab_raw(HighFive::DataSet ds,
```

Comment:
I think this can be modelled in HighFive by creating a `Selection` and then writing it. Selections also have a `write_raw` function.

I don't think it's important to change the code – I assume the current code works – but it might be worthwhile to add a respective comment for future reference.

Reply:
I played with this for a bit yesterday and again just now -- HighFive doesn't seem to have a `write` or `write_raw` that works for all of the types/ here,
https://github.com/kotekan/kotekan/blob/83e205688fe1a079d1a6c0feb01e8bb9b5bbdfbc/lib/utils/visFileArchive.cpp#L155
There might be a way around this though.